modules/jsonld/src/main/scala/ch/epfl/bluebrain/nexus/rdf/jsonld/JsonLd.scala

```diff
@@ -0,0 +1,118 @@
+package ch.epfl.bluebrain.nexus.rdf.jsonld
+
+import cats.Monad
+import cats.data.EitherT
+import ch.epfl.bluebrain.nexus.rdf.Iri
+import ch.epfl.bluebrain.nexus.rdf.Iri.AbsoluteIri
+import io.circe.Json
+import io.circe.syntax._
+import cats.implicits._
+
+object JsonLd {
+
+  /**
+    * Resolve all IRIs inside @context.
+    *
+    * @param json     [[Json]] with context to resolve
+    * @param resolver context resolver
+    * @return [[Json]] with all the context IRIs resolved
+    */
+  def resolveContext[F[_]: Monad](
+      json: Json
+  )(resolver: AbsoluteIri => F[Option[Json]]): EitherT[F, ContextResolutionException, Json] = {
+
+    def inner(resolvedIds: List[AbsoluteIri], context: Json): EitherT[F, ContextResolutionException, Json] =
```

Comment:
Is there a reason why resolvedIds has to be a `List`. A `Set` should be more performant

Reply:
We need it, because we need the order in which contexts for displaying circular dependencies